eesp-ikev2.org

```diff
@@ -82,154 +83,76 @@ described in BCP 14 [[RFC2119]] [[RFC8174]] when, and only when, they
-
-** Implicit Initialization Vectors
-
-With the Implicit Initialization Vector (IIV) encryption algorithm,
-as specified in [[RFC8750]], the IV MUST be omitted in the EESP
-packet. To enable this functionality, IIV transforms defined in
-[[IKEv2-Enc]] MUST be used during negotiation. Furthermore,
-the [[IKEv2-SN]] extension MUST be negotiated to support the use of
-64-bit Sequential Numbers in EESP packets. If the the proposal
-does not include 64-bit Sequential Numbers return error
-NO_PROPOSAL_CHOSEN.
+# [VS] See note above. This should either be a normative reference
+# [VS] And we use a different term - Sub SA. "Sub-Child SA" is not
+# [VS] used in the text. I'd rather to delete this
+# This document uses the following terms defined in
+# [[I-D.mrossberg-ipsecme-multiple-sequence-counters]]: Sub-Child SA.
+
+# [VS] This is the name of transform, I don't think we should 
+# [VS] reference ikev2-rename-esn here
+# [VS] ikev2-rename-esn is only relevant *now* until it becomes an 
+# [VS] RFC and IANA updates IKEv2 registries
+# [VS] For this reason it is referenced in G-IKEv2 draft, which uses 
+# [VS] not-yet-assigned name for this transform
+# [VS] But by the time this draft would be published, the 
+# [VS] ikev2-rename-esn will most probably become an RFC and IANA 
+# [VS] completes the renaming,
+# [VS] thus we can just reference the IANA registry
+# This document uses the following terms defined in
+# [[I-D.ietf-ipsecme-ikev2-rename-esn]] : Replay Protection.
+
+# [VS] I'd rather to put off all the group SA stuff 
+# This document uses the following terms defined in
+# [[I-D.ietf-ipsecme-g-ikev2]]: Sender-ID, Data-Security SA,
+# GWP_SENDER_ID_BITS, GCKS policy.
+
+* EESP Overview
 
 ** EESP Version
+
 Each EESP packets carry EESP Base Header version, which is specified
-[[I-D.klassert-ipsecme-eesp]]. This SHOULD BE negotiated using
-IKEv2. Each Base Header version, to be able to negotiate via IKEv2,
-SHOULD have a corresponding ~IKEv2 Security Protocol Identifiers~
-The initial version sepecified EESPv0(TBD1)
+in Section [XXX] of [[I-D.klassert-ipsecme-eesp]]. This SHOULD be 
+negotiated using IKEv2. Each Base Header version, to be able to 
+negotiate via IKEv2, SHOULD have a corresponding ~IKEv2 Security 
+Protocol Identifiers~ The initial version sepecified EESPv0 (TBD1)
 
-* Sub SA
+** EESP Sub SA
 Existing mechanisms for establishing Child SAs, as described in
 [[RFC7296]], yield pair of SAs. High-speed IP traffic is often
 asymmetric. Creating multiple pairs of Child SAs, e.g. [[RFC9611]] or
-for DSCP to carry asymetric traffic is inefficient. A pair of Child SA
-combined with multiple unidirectional a set of  Sub SAs with Sub SA
-IDs greater than 0, provides a more flexible approach to carrying
+for DSCP to carry asymetric traffic is inefficient. A pair of Child 
+SA combined with multiple unidirectional a set of Sub SAs with Sub 
+SA IDs greater than 0, provides a more flexible approach to carrying
```

Comment:
Similar to my comment above, I feel the term Child SA is not correctly used here. It's a pair of EESP SAs that are part of a single Child SA. And it's unclear to me what "combined with multiple unidirectional a set of" is supposed to mean.

Reply:
Hmm, this is an interesting question. I've been always thinking that Child SA is a generic name for IPsec SA (so that a ESP SA is a Child SA and an AH SA is a Child SA). Now I re-read RFC 7296 and I can see a support for your interpretation - that Child SA is a pair of ESP SAs or a pair of AH SAs. But there are also places in RFC 7296 that support my interpretation, like Section 2.8.1 that states: "At this point, there are three Child SA pairs between A and B (the old one and two new ones)" - in this case it counts an uni-direction ESP or AH SA as a Child SA.
Thus RFC 7296 is vague about this :-)